core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+<script type="text/ng-template" id="components/topNavigationBar">
```

Comment:
Understood, and I agree that this is how we might want to do it if we decided to split up the top navbar. However, I am actually quite happy with the current directive as a single unit, and I don't really see a need to split it up further at the moment. If we want to introduce a standard that directives should not be > N lines in length, I suggest that we look into this in a separate pass, but I think that there's also some conceptual simplicity in having the top navbar in a single file (within reason, of course).